src/server/models/pushSubscriptionModel.ts

```diff
@@ -10,48 +12,53 @@ interface PushSubscriptionInfo {
 }
 
 class PushSubscriptionModel {
-  private pushSubscriptions: {[id: string]: PushSubscriptionInfo[]};
+  private firestore: Firestore;
 
   constructor() {
-    this.pushSubscriptions = {};
+    this.firestore = new Firestore();
+  }
+
+  _getSubscriptionCollection(login: string) {
+    const collectionPath = `pushsubscriptions/${login}/subscriptions`;
```

Comment:
How about we call the top-level collection `users`? Seems redundant to have `pushsubscriptions > $login > subscriptions`, but `users > $login > subscriptions` makes sense. Then we can shove other user info in the same user documents.